src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirModel.cs

```diff
@@ -80,88 +77,121 @@ public sealed class SqlServerFhirModel : IStartable
         }
 
         public byte GetCompartmentTypeId(string compartmentType)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
             return _compartmentTypeToId[compartmentType];
         }
 
         public bool TryGetSystemId(string system, out int systemId)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
             return _systemToId.TryGetValue(system, out systemId);
         }
 
         public int GetSystemId(string system)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
 
             VLatest.SystemTable systemTable = VLatest.System;
             return GetStringId(_systemToId, system, systemTable, systemTable.SystemId, systemTable.Value);
         }
 
         public int GetQuantityCodeId(string code)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
 
             VLatest.QuantityCodeTable quantityCodeTable = VLatest.QuantityCode;
             return GetStringId(_quantityCodeToId, code, quantityCodeTable, quantityCodeTable.QuantityCodeId, quantityCodeTable.Value);
         }
 
         public bool TryGetQuantityCodeId(string code, out int quantityCodeId)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
             return _quantityCodeToId.TryGetValue(code, out quantityCodeId);
         }
 
-        // TODO: Once this no longer implements IStartable, rename to "EnsureInitialized" (work item 75558).
-        public void Start()
+        public Task EnsureInitialized()
         {
-            _schemaInitializer.Start();
-
-            if (_searchParameterDefinitionManager is IStartable startable)
+            if (_schemaInformation.Current == null)
             {
-                startable.Start();
+                // TODO: What is the correct error to throw here?
+                throw new NullReferenceException("The current schema version should not be null");
             }
 
-            // As long as SqlServerFhirModel implements IStartable, avoid initialization if we are only running the base schema.
-            if (_schemaInformation.Current == null)
+            // If the fhir-server is just starting up, synchronize the fhir-server dictionaries with the SQL database
+            Initialize((int)_schemaInformation.Current, true);
+
+            return Task.CompletedTask;
+        }
+
+        public void Initialize(int version, bool runAllInitialization)
+        {
+            if (_highestInitializedVersion == version)
             {
                 return;
             }
 
             var connectionStringBuilder = new SqlConnectionStringBuilder(_configuration.ConnectionString);
+            _logger.LogInformation("Initializing {Server} {Database} to version {Version}", connectionStringBuilder.DataSource, connectionStringBuilder.InitialCatalog, version);
+
+            if (runAllInitialization)
+            {
+                InitializeV1();
```

Comment:
I think I would give this a more descriptive name, maybe something like `InitializeBase`, since this name is a bit misleading since it has to be run even for V>1 of the schema.

Similarly, maybe give InitializeV6 a name like `InitializeSearchParamStatuses`

Reply:
Ah yes, that sounds much better. Looks like both you and @Ivanidzo4ka are on the same page. Renamed the methods in eef506a 👍